### PR TITLE
MGMT-7105: add AI vsphere platform CI

### PR DESCRIPTION
**Pull request by michaellevy101**
As a first step, this PR aims to:
Create a workflow that has access to the vsphere environment and creates a Centos8 template.
As an alternative to the packet baremetal server, this template will be cloned for the assisted-test-infra machine.

### CODE REVIEW

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml
Comment (osherdp) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:388]: why's that change?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml
Comment (michaellevy101) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:388]: a wrong copy paste. 
Please don't review this PR yet, it isn't ready

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml:1250]: this means it will run for each PR in assisted-test-infra, while still not being required for the PR to merge
I'm okay with that, just wanted to make sure you're aware

File: ci-operator/step-registry/assisted/vsphere/assisted-vsphere-workflow.yaml
Comment (osherdp) [inline ci-operator/step-registry/assisted/vsphere/assisted-vsphere-workflow.yaml:12]: please remove commented-out code

File: ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml
Comment (osherdp) [inline ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml:6]: same

File: ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml
Comment (osherdp) [inline ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml:4]: cluster-bot? what does it have to do with this workflow?

File: ci-operator/step-registry/assisted/vsphere/setup/template/assisted-vsphere-setup-template-ref.yaml
Comment (osherdp) [inline ci-operator/step-registry/assisted/vsphere/setup/template/assisted-vsphere-setup-template-ref.yaml:11]: where do we use those ibmcloud creds?

File: ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml
Comment (osherdp) [inline ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml:9]: the assisted-service itself gets installed on vsphere?
can you explain in the PR description how it happens?

**Review by osherdp — CHANGES_REQUESTED** [review]
partial review

File: ci-operator/step-registry/assisted/OWNERS
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/OWNERS:1]: right, don't want to break things as part of this effort can be in a different PR

File: ci-operator/step-registry/assisted/OWNERS
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/OWNERS:3]: ok. didn't know

Comment (michaellevy101) [reply ci-operator/step-registry/assisted/OWNERS]: removed all redundant occurrences

File: ci-operator/step-registry/assisted/vsphere/setup/template/assisted-vsphere-setup-template-ref.yaml
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/template/assisted-vsphere-setup-template-ref.yaml:11]: We have two vsphere envs on AWS and on IBM Cloud.
The LEASED_RESOURCE controls what cluster to use.
please read: https://github.com/openshift/release/blob/6fcfd06ade2161c7dad325778072538ac9c0855a/ci-operator/step-registry/ipi/conf/vsphere/check/ipi-conf-vsphere-check-commands.sh#L24

File: ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml:9]: This workflow should install the assisted service on vsphere but this PR is only the first step, creating a template for the test-infra machine instead of packet server

File: ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml]: It used for Openshift for testing but can be eliminated

File: ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/assisted-vsphere-setup-chain.yaml:5]: documented

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml
Comment (michaellevy101) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml:1250]: This configuration is auto generated by "make jobs" command. 
For some reason "make update" didn't update this code as expected.
running "make jobs" manually did the trick. nice catch.

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/assisted/OWNERS
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/OWNERS:1]: It a huge change and i don't see a good reason to mix it all together.
It requires changing filenames and steps names and have so much places to make mistakes.
My goal is to create a vsphere CI not to refactor everything for nothing but a folder structure. It's a classical technical dept.

Comment (eliorerz) [reply ci-operator/step-registry/assisted/OWNERS:1]: I'm agreeing with @osherdp on this. If you are creating a directory for `assisted` then baremetal should be there too.
To my opinion, if you want to change baremetal root hierarchy in different PR, its should happen before this one and not add it next to the baremetal directory

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml
Comment (eliorerz) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:377]: Maybe create a periodic for this job so we can monitor the failures (if there will be any)

**Review by osherdp — APPROVED** [review]

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml
Comment (michaellevy101) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:377]: will do,  after finishing the development (currently we test nothing just creating a template)

**Review by rvanderp3 — COMMENTED** [review]
Is the intent for this PR to just get the job established and installing the centos VM?  Will there be a follow on PR to install an OpenShift cluster?

File: ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh:1]: It's not duplicate since this step use different user (AI CI users instead of taking a random user from a pool).

File: ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh:13]: @rvanderp3 thanks for reviewing.
In contrast, the vsphere storage and networking configuration should not be included in the logic. It doesn't make sense to create and approve a PR if we want to change something (such as changing the datastore). 
we can resolve this issue by creating a solution for configuration like config map on kubernetes.

File: ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh
Comment (rvanderp3) [reply ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh:1]: This makes sense. The convention, atleast for vSphere, has been to keep the configuration in the release repo.  But given that you'll be maintaining this workflow, I have no issue with the way you've done it here.

File: ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh
Comment (rvanderp3) [reply ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh:14]: One of the benefits of tracking the changes via PR is the visibility of changes and a tracked history.  We have been keeping the configuration in the release repo since we started testing CI on vSphere as a way to manage and update the configuration of jobs.  Yes, I and the rest of the maintainers of the vSphere CI environment understand this repository is public.  Though it never hurts to be reminded of that :) .  As I mentioned above, I'm not opposed to the way you've done this, it's just in contrast to the way other vSphere jobs are composed.

**Comment by rvanderp3** [comment]
@michaellevy101 thanks for the responses above, just let me know when you're ready for an lgtm.

File: ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh
Comment (michaellevy101) [reply ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh:14]: thanks @rvanderp3:)

**Comment by michaellevy101** [comment]
> @michaellevy101 thanks for the responses above, just let me know when you're ready for an lgtm.

I'm ready. thanks!!!!

File: ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh
Comment (rvanderp3) [reply ci-operator/step-registry/assisted/vsphere/setup/configure/assisted-vsphere-setup-configure-commands.sh:14]: no problem at all @michaellevy101 :)

**Pull request merged**
